src/v2/System/useFeatureFlag.tsx

```diff
@@ -13,6 +22,34 @@ export function useFeatureFlag(flagName: string): boolean | null {
   }
 
   return (
-    Object.keys(featureFlags).indexOf(flagName) !== -1 && featureFlags[flagName]
+    Object.keys(featureFlags).indexOf(flagName) !== -1 &&
+    featureFlags[flagName].flagEnabled
   )
 }
+
+export function useFeatureVariant(featureName: string): Variant {
+  const { featureFlags } = useSystemContext()
+  const errorObject = {
+    enabled: false,
+    name: "false",
+  }
+
+  if (featureFlags === undefined) {
+    console.error(
+      "[Force] Error: featureFlags is undefined in SystemContext in UseFeatureVariant"
+    )
+    return errorObject
+  }
+
+  if (!featureName) {
+    console.error("[Force] Error: no argument passed into useFeatureVariant")
+    return errorObject
+  }
+
+  if (!featureFlags[featureName]) {
+    console.error("[Force] Error: the variant can't be found on featureFlags")
+    return errorObject
+  }
```

Comment:
Similar to feedback on a previous PR, each of these error cases could be a bit of overkill:

- `featureFlags` existy check not necessary as the default can be an empty object within system context. `featureFlags` will always exist
- `featureName` existy will 100% of the time be caught by TypeScript
- `!featureFlags[featureName]` is a valid check

Really all we need (both here and in `useFeatureFlag` is  

```tsx
const useVariant(featureName:string) => {
  const { featureFlags } = useSystemContext()
  const variant = featureFlags[featureName]?.variant
  
  if (!variant) {
    console.error("[Force]: Error: the variant can't be found on featureFlags", featurefFlags) 
    return null 
  }
  
  return variant
}
```

We should avoid stubbing out mock flag responses for error states as the consumer of the flag should be the one checking for `null`, and then handling it there.

Reply:
(Referring to bullet point #1, [this line](https://github.com/artsy/force/blob/main/src/desktop/lib/buildServerAppContext.ts#L28) (and corresponding line in buildClientAppContext) can be:

```tsx
featureFlags: res.locals.sd.FEATURE_FLAGS ?? {}
```

or even better, and more lower down so we don't have to update these context files,
```tsx
res.locals.sd.FEATURE_FLAGS = <some logic to set> ?? {} 
```

Reply:
>or even better, and more lower down so we don't have to update these context files

This should be handled here: https://github.com/artsy/force/blob/main/src/lib/middleware/featureFlagMiddleware.ts#L32

Reply:
oh sweet 🙏